src/ring_theory/multiplicity.lean

```diff
@@ -57,6 +57,18 @@ begin
     apply _root_.le_antisymm; { apply nat.find_mono, norm_cast, simp } }
 end
 
+example (a b : ℤ) : a ∣ b ↔ a ∣ b.nat_abs := int.dvd_nat_abs.symm
+
+theorem nat_cast_nat_abs (a b : ℕ) :
+    multiplicity a (b : ℤ).nat_abs = multiplicity (a : ℤ) (b : ℤ) :=
+begin
+  apply part.ext',
+  { repeat { rw [← finite_iff_dom, finite_def] },
+    norm_cast },
+  { intros h1 h2,
+    apply _root_.le_antisymm; { apply nat.find_mono, norm_cast, simp } }
+end
+
```

Comment:
This lemma is strictly superseded by the other lemma that you add below. And you don't need this lemma in the proof there.
```suggestion
```